starstream-to-wasm/src/lib.rs

```diff
@@ -248,12 +258,25 @@ impl Compiler {
                     func.instructions().i32_eqz();
                     func.instructions().br_if(1);
                     // contents
-                    self.visit_block(func, &(parent, &locals), &body.statements);
+                    self.visit_block(func, &(parent, &locals), body);
                     // continue
                     func.instructions().br(0).end().end();
                 }
+                TypedStatement::Return(Some(expr)) => {
+                    let im = self.visit_expr(func, &(parent, &locals), expr.span, &expr.node);
+                    self.discard(func, im);
```

Comment:
Future work: conform expr to function return type rather than unconditionally discarding